### PR TITLE
update readme with details on config fields, behavior, troubleshooting

### PR DESCRIPTION
**Pull request by gabemontero**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,100 @@
...
+
+The samples operator’s image contains the imagestream and template definitions for the associated OpenShift release.  It uses annotations in each sample to denote the version, and the operator makes sure that any samples in the openshift namespace that it has content for its associated version by comparing the annotation with its release level.  Samples outside of its inventory are ignored, as are skipped samples (see below).   It will watch the imagestreams and templates in the openshift namespace and correct modifications to managed samples if needed.
+
+The sample resource is a cluster wide resource that employs a finalizer to allow for deletion of the samples resource to also delete the imagestreams and templates in the openshift namespace, any openshift-sample secret copied into the openshift namespace, as well as any generated config resources or cluster status resources in the operator’s namespace.
+
+Upon deletions of the samples resource, the samples operator will create the default samples resource again.
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply README.md]: part of next push

File: README.md [hunk 2]
```diff
@@ -1,8 +1,100 @@
...
-The samples operator installs+maintains the sample templates+imagestreams on a cluster
-
-## To update the content in this repository from https://github.com/openshift/library
-1. Create a git branch in you local copy of this repository
-2. cd tmp/build
-3. ./library-sync.sh
-4. Commit the changes and create a PR
+# Overview
+
+The samples operator manages the sample imagestreams and templates stored in the openshift namespace in the cluster, and any docker credentials, stored in secret, needed for the imagestreams to import the images they reference.
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply README.md]: part of next push

File: README.md [hunk 3]
```diff
@@ -1,8 +1,100 @@
...
+
+On initial startup, the operator will create the default samples resource to initiate the creation of the imagestreams and templates.  At this moment, the imagestreams are the centos based OKD imagestreams pointing to images on docker.io.  And similarly the template are those categorized as OKD templates.  When details around getting the proper credentials by default for registry.redhat.io are sorted out, we’ll most likely switch to defaulting to the rhel and OCP based content.
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply README.md]: part of next push

File: README.md [hunk 4]
```diff
@@ -1,8 +1,100 @@
...
+
+The samples operator runs in the openshift-cluster-samples-operator namespace.  Configuration resources (current state or human supplied secret to us for registry access) also reside in that namespace.  If an admin supplies a docker config json file formatted secret with the name “openshift-samples” in the operator’s namespace, it will be copied into the openshift namespace for use during imagestream import.
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply README.md]: part of next push

File: README.md [hunk 5]
```diff
@@ -1,8 +1,100 @@
...
+
+The samples operator’s image contains the imagestream and template definitions for the associated OpenShift release.  It uses annotations in each sample to denote the version, and the operator makes sure that any samples in the openshift namespace that it has content for its associated version by comparing the annotation with its release level.  Samples outside of its inventory are ignored, as are skipped samples (see below).   It will watch the imagestreams and templates in the openshift namespace and correct modifications to managed samples if needed.
+
+The sample resource is a cluster wide resource that employs a finalizer to allow for deletion of the samples resource to also delete the imagestreams and templates in the openshift namespace, any openshift-sample secret copied into the openshift namespace, as well as any generated config resources or cluster status resources in the operator’s namespace.
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply README.md]: part of next push

File: README.md [hunk 6]
```diff
@@ -1,8 +1,100 @@
...
+Upon deletions of the samples resource, the samples operator will create the default samples resource again.
+
+# Configuration
+
+The samples resource offers the following configuration fields:
+
+- ManagementState
+-- Managed: the operator will update the samples as the configuration dictates
+-- Unmanaged: the operator will ignore updates to the samples resource object and any imagestreams or templates in the openshift namespace
+-- Removed: the operator will remove the set of managed imagestreams and templates in the openshift namespace. It will ignore new samples created by the cluster admin or any samples in the skipped lists.  After the removals are complete, the operator will work like it is in Unmanaged state and ignore any watch events on the sample resources, secrets, imagestreams, or templates.  There are some caveats around concurrent creates and removal (see Change behaviors section).
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply README.md]: I think `in the Unmanaged state` is better in this case ... will switch to that

File: README.md [hunk 7]
```diff
@@ -1,8 +1,100 @@
...
+- Architecture
+-- Place holder to choose between x86 and ppc content
+- Skipped imagestreams
+-- Imagestreams that are in the operator’s inventory, but that the cluster admin wants the operator to ignore / not manage
+- Skipped templates 
+-- Templates that are in the operator’s inventory, but that the cluster admin wants the operator to ignore / not manage
+
+## Config restrictions
+
+The install type and architectures are not allowed to be changed once the initial startup and creation has occurred.
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply README.md]: part of next push

**Comment by gabemontero** [comment]
Updates from @coreydaley 's first iteration of comments pushed.

File: README.md [hunk 8]
```diff
@@ -1,8 +1,110 @@
...
+The samples resource offers the following configuration fields:
+
+- ManagementState
+-- Managed: the operator will update the samples as the configuration dictates
+-- Unmanaged: the operator will ignore updates to the samples resource object and any imagestreams or templates in the openshift namespace
+-- Removed: the operator will remove the set of managed imagestreams and templates in the openshift namespace. It will ignore new samples created by the cluster admin or any samples in the skipped lists.  After the removals are complete, the operator will work like it is in the 'Unmanaged' state and ignore any watch events on the sample resources, secrets, imagestreams, or templates.  There are some caveats around concurrent creates and removal (see Change behaviors section).
+- Samples Registry
+-- Override the registry that images are imported from
+- Install type
+-- Use 'centos' for CentOS imagestreams and 'rhel' for RHEL imagestreams (and associatively between OCP and OKD content for the templates)
```
Comment (bparees) [inline README.md]: s/OCP and OKD/OKD and OCP/  (keep the ordering consistent....  centos/rhel and okd/ocp)

**Comment by bparees** [comment]
@gabemontero did you see my two comments?  if you fix those and squash i'll just manually merge this since there's no code change.

**Comment by gabemontero** [comment]
I missed them in the flurry of activity @bparees ... thx for reminder.

I'm in the middle of debugging some issues my skipped e2e tests have uncovered, but will circle back to this on Monday - thanks

File: README.md (hunk 8 above)
Comment (gabemontero) [reply README.md]: part of next push

**Comment by gabemontero** [comment]
updates pushed @bparees and commits squashed

**Pull request merged**